### DISCUSSION TITLE
Force control on wrists

### DISCUSSION THREAD

**Discussion by loreucci** [robotology/community#367] [Q&A] [answered]
Hi,

we are trying to use force control on some joints of the iCub arm, including at least one wrist joint. We followed the instructions on the manual (http://wiki.icub.org/wiki/Force_Control), but there is a TODO in the part related to the activation of force control for the wrist.

Is it possible to achieve this? And if so, how could we enable force control on the wrists?

Thank you in advance for any help.

**Comment by pattacini** [comment, marked as answer]
Hi @loreucci 

We recommend not using torque control on the wrist for the reasons detailed in the wiki. Further, we will update the documentation accordingly, removing the TODO section.

However, enabling torque control on the first 5 joints of the arm is practically always resolutive for doing many tasks, at least in our experience.

What specific goal do you have in mind that foresees the use of a torque-controlled wrist?

**Comment by loreucci** [comment]
Hi @pattacini, thank you for the reply.

We are investigating the possibility of performing a throwing movement with the iCub arm, using shoulder_pitch, elbow and wrist_pitch joints. We were thinking of using torque control as it would give us the possibility of performing more dynamics movements. We did some preliminary tests in Gazebo, but when we moved on the real robot we encountered this issue.

Do you think it's feasible to achieve torque control for wrist_pitch?

**Comment by pattacini** [comment]
@loreucci I have no hands-on experience for this. Anyway, given that the wrist actuation is based on a differential drive, you ought to keep the wrist pitch and yaw always on the same control mode.

Besides that, does anyone with the right know-how who's more confident than me on the wrist want to chime in and comment further? Perhaps @randaz81 or a member of @DanielePucci's group?

**Comment by DanielePucci** [comment]
CC @traversaro

**Comment by pattacini** [comment]
cc @robotology/iit-dic

**Comment by traversaro** [comment]
> Do you think it's feasible to achieve torque control for wrist_pitch?

Note: the following is just my personal intuition, and it may be contradicted after performing experiments on the real robot. I suspect that torque_control for wrist_pitch would be quiet tricky, as the estimated torque in the wrist was quite noisy. Furthermore, even if you had good torque feedback the tendon actuation introduces a bit of elasticity that may not be trivial to deal with when using torque control on that joint. In general it may be possible to do torque control for the wrist_pitch, but I suspect it would require tuning the parameters and eventually modifying both the torque estimation algorithm, and the gain of the low-level torque controller.

**Comment by loreucci** [comment]
Thank you @traversaro for the reply.

We got the idea that it is really cumbersome to use torque control on the wrist_pitch joint and we are trying to avoid this and instead rely on velocity control, for the same task. It seems that we can more or less achieve the same goal in this manner.

Thank you all again for the support.
